idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -0,0 +1,1799 @@
+
+        self.vapor_phase.add_material_balances(
+            balance_type=self.config.vapor_side.material_balance_type,
+            has_phase_equilibrium=self.config.vapor_side.has_phase_equilibrium,
+            has_mass_transfer=self.config.vapor_side.has_mass_transfer)
+
+        self.vapor_phase.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_transfer=self.config.vapor_side.has_heat_transfer)
+
+        self.vapor_phase.add_momentum_balances(
+            balance_type=self.config.vapor_side.momentum_balance_type,
+            has_pressure_change=self.config.vapor_side.has_pressure_change)
+
+        self.vapor_phase.apply_transformation()
+
+    # ==========================================================================
+        """ Build Control volume 1D for liquid phase and
+            populate liquid control volume"""
+        self.liquid_phase = ControlVolume1DBlock(default={
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.liquid_side.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points,
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.liquid_side.property_package,
+            "property_package_args":
+                self.config.liquid_side.property_package_args})
+
+        self.liquid_phase.add_geometry(flow_direction=set_direction_liquid,
+                                       length_domain_set=self.config.
+                                       length_domain_set)
+
+        self.liquid_phase.add_state_blocks(
+            information_flow=set_direction_liquid,
+            has_phase_equilibrium=False)
+
+        self.liquid_phase.add_material_balances(
+            balance_type=self.config.liquid_side.material_balance_type,
+            has_phase_equilibrium=self.config.liquid_side.has_phase_equilibrium,
+            has_mass_transfer=self.config.liquid_side.has_mass_transfer)
+
+        self.liquid_phase.add_energy_balances(
+            balance_type=self.config.energy_balance_type,
+            has_heat_transfer=self.config.liquid_side.has_heat_transfer)
+
+        self.liquid_phase.apply_transformation()
+
+        # Add Ports for vapor side
+        self.add_inlet_port(name="vapor_inlet", block=self.vapor_phase)
+        self.add_outlet_port(name="vapor_outlet", block=self.vapor_phase)
+
+        # Add Ports for liquid side
+        self.add_inlet_port(name="liquid_inlet", block=self.liquid_phase)
+        self.add_outlet_port(name="liquid_outlet", block=self.liquid_phase)
+
+    # ==========================================================================
+        """ Add performace equation method"""
+        self._make_performance()
+
```

Comment:
I like to put them in methods so that I can easily fold (expand) and unfold them. In this case, I have the performance equations in a method, the initialization method and so on.